snow/networking/benchlist/benchlist.go

```diff
@@ -131,11 +148,22 @@ func (b *queryBenchlist) QueryFailed(validatorID ids.ShortID, requestID uint32)
 		return
 	}
 
+	// Track the message failure and bench [validatorID] if it has
+	// surpassed the threshold
 	key := validatorID.Key()
-	// Add a failure and benches [validatorID] if it has
-	// passed the threshold
-	b.consecutiveFailures[key]++
-	if b.consecutiveFailures[key] >= b.threshold {
+	currentTime := b.clock.Time()
+	failureStreak := b.consecutiveFailures[key]
+	failureStreak.consecutive++
+	if failureStreak.firstFailure.IsZero() {
+		failureStreak.firstFailure = currentTime
+		b.consecutiveFailures[key] = failureStreak
+		// Return early here since a peer will not
+		// be benched after its first failure
+		return
+	}
```

Comment:
Minor nits... But I think it would be cleaner for this to be more along the lines of:
```
failureStreak := b.consecutiveFailures[key]
if failureStreak.consecutive == 0 {
    failureStreak.firstFailure = currentTime
}
failureStreak.consecutive++
```
It removes some code and handles the case that `b.threshold = 1` (which would be super aggressive, but seems like something that would make sense to work here)